hazelcast/src/main/java/com/hazelcast/map/impl/wan/WanMapEntryView.java

```diff
@@ -18,24 +18,34 @@
 
 import com.hazelcast.core.EntryView;
 import com.hazelcast.internal.nio.IOUtil;
+import com.hazelcast.internal.serialization.Data;
+import com.hazelcast.internal.serialization.SerializationService;
+import com.hazelcast.internal.serialization.SerializationServiceAware;
 import com.hazelcast.nio.ObjectDataInput;
 import com.hazelcast.nio.ObjectDataOutput;
 import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
 import com.hazelcast.wan.impl.WanDataSerializerHook;
 
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 import java.io.IOException;
 import java.util.Objects;
 
+import static com.hazelcast.internal.util.Preconditions.checkNotNull;
+
 /**
  * WAN heap based implementation of {@link EntryView}.
  *
  * @param <K> the type of key.
  * @param <V> the type of value.
  */
-public class WanMapEntryView<K, V> implements EntryView<K, V>, IdentifiedDataSerializable {
-
+@SuppressWarnings("checkstyle:methodcount")
+public class WanMapEntryView<K, V> implements EntryView<K, V>, IdentifiedDataSerializable, SerializationServiceAware {
```

Comment:
maybe we can have a test for this class?

Reply:
Sure.